engines/python/setup/djl_python/utils.py

```diff
@@ -27,9 +27,9 @@ class InputFormatConfigs:
     tokenizer: Any = None
 
 
-def parse_input_with_formatter(
-        inputs: Input,
-        input_format_configs: InputFormatConfigs) -> ParsedInput:
+def parse_input_with_formatter(inputs: Input,
+                               input_format_configs: InputFormatConfigs,
+                               adapter_registry: dict) -> ParsedInput:
```

Comment:
I don't think conceptually `adapter_registry` fits well within `InputFormatConfigs` as it's not a config. 

I agree that `adapter_registry` arg should be an optional arg, good point.

Reply:
> Can adapter_registry be a part of InputFormatConfigs? The reason for input format configs, is in the future we would want to make this a public API and we dont want the number of parameters to grow here.
> 
> Also, adapter_registry optional with default value empty dict, since tensorrt llm handler does not support adapters yet.

"in the future we would want to make this a public API" -- public to whom? Can't we address that "in the future" then? And doesn't defaulting to `{}` basically solve that concern here?

If the number of parameters starts to grow and the API being public becomes a concern, can't we create some kind of "Context" container or something to hold these parameters at that time?

Reply:
> "in the future we would want to make this a public API" -- public to whom? Can't we address that "in the future" then? 
> 
To people who are writing custom pre/post processing in their custom handler. We have output_formatter as Callable function for output formatting, we would want the same thing for input parsing as well, as our customers are asking for it.  

> And doesn't defaulting to `{}` basically solve that concern here?
>
Yes, but we could UX could be better. We dont have to figure it today.